frontend/src/entities/group/create/group-create.tsx

```diff
@@ -0,0 +1,217 @@
+    const notificationService = NotificationService(dispatch);
+    const navigate = useNavigate();
+    const {groupName} = useParams();
+    const [initialLoaded, setInitialLoaded] = useState(false);
+
+    const groupSchema = z.object({
+        name: z
+            .string()
+            .min(3, {
+                message: 'Group name must be longer than 3 characters',
+            })
+            .max(24, {
+                message: 'Group name cannot be more than 24 characters',
+            })
+            .regex(/^[a-zA-Z0-9]+$/, {
+                message: 'Group name can only contain alphanumeric characters.',
+            }),
+        description: z
+            .string({
+                invalid_type_error: 'Group description must be a string',
+            })
+            .min(1, {
+                message: 'Group description cannot be empty',
+            })
+            .max(4000, {
+                message: 'Group description cannot be more than 4000 characters.',
+            })
+            .regex(/[ -~]/, {
+                message: 'Group description can only contain printable characters',
+            }),
+    });
+
+    const {state, setState, setFields, touchFields} = useValidationReducer(groupSchema, {
+        validateAll: false,
+        needsValidation: false,
+        form: {
+            name: '',
+            description: '',
+        },
+        touched: {},
+        formErrors: {} as any,
+        formSubmitting: false,
+    });
+
+    useMemo(() => {
+        if (isEdit && !initialLoaded) {
+            setState({formSubmitting: true});
+            dispatch(getGroup(groupName)).then((response) => {
+                if (response.payload) {
+                    setInitialLoaded(true);
+                    setState({
+                        form: {
+                            name: response.payload?.data.group?.name || '',
+                            description: response.payload?.data.group?.description || '',
+                        },
+                        formSubmitting: false
+                    });
+                } else {
+                    navigate('/404');
+                }
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isEdit, groupName, initialLoaded]);
+
+
+    state.formErrors = {};
+    const parseResult = groupSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        state.formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
+
+    async function handleSubmit () {
+        setState({formSubmitting: true});
+        let response = null;
+        try {
+            if (!isEdit) {
+                response = await dispatch(createGroup(state.form));
+            } else {
+                response = await dispatch(updateGroup(state.form));
+            }
+            if (response.payload?.status === 200) {
```

Comment:
So we don't have to do a bunch of explicit status checks everywhere we've been trying to standardize on [isFulfilled](https://redux-toolkit.js.org/api/matching-utilities#isfulfilled). You can see an [example here](https://github.com/awslabs/mlspace/blob/65547456b15f42268766527eba88cf0e26365ab1/frontend/src/entities/notebook/notebook.actions.tsx#L169-L174)

Reply:
So it seems isFulfilled just validates that the thunkRequest finished and we will need to keep this logic or add logic in the reducer to check for response statuses and reject appropriately. I tested this and then navigated to edit a non-existent group, the request returned a 404 but isFulfilled still returned `true`.